wallet/datastore.go

```diff
@@ -431,47 +354,419 @@ func (pg *Postgres) IncreaseLinkingLimit(ctx context.Context, providerLinkingID
+	if err != nil {
+		return fmt.Errorf("failed to create db transaction ConnectCustodialWallet: %w", err)
+	}
+	// will rollback if tx created at this scope
+	defer rollback()
+
+	var existingLinkingID uuid.UUID
+	// get the custodial provider's linking id from db
+	stmt := `
+		select linking_id from wallet_custodian
+		where wallet_id=$1 and custodian=$2
+	`
+	err = tx.Get(&existingLinkingID, stmt, cl.WalletID, cl.Custodian)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		sublogger.Error().Err(err).
+			Msg("failed to insert wallet_custodian")
+		return fmt.Errorf("failed to insert wallet custodian record: %w", err)
+	}
+
+	if !uuid.Equal(existingLinkingID, *new(uuid.UUID)) {
+		// check if the member matches the associated member
+		if !uuid.Equal(*cl.LinkingID, existingLinkingID) {
+			return handlers.WrapError(errors.New("wallets do not match"), "unable to match wallets", http.StatusForbidden)
+		}
+	}
+
+	// get the count
+	used, max, err := pg.GetCustodianLinkCount(ctx, *cl.LinkingID)
+	if err != nil {
+		sublogger.Error().Err(err).
+			Msg("failed to insert wallet_custodian due to db err checking linking limits")
+		return fmt.Errorf("failed to insert wallet custodian record due to db err checking linking limits: %w", err)
+	}
+
+	// check for linking limit
+	if used >= max {
 		sentry.WithScope(func(scope *sentry.Scope) {
-			anonAddr := ""
-			if anonymousAddress != nil {
-				anonAddr = anonymousAddress.String()
-			}
 			scope.SetTags(map[string]string{
-				"walletId":               ID,
-				"providerLinkingId":      providerLinkingID.String(),
-				"anonymousAddress":       anonAddr,
-				"userDepositDestination": userDepositDestination,
+				"wallet_id":  cl.WalletID.String(),
+				"linking_id": cl.LinkingID.String(),
 			})
 			tooManyCardsCounter.Inc()
 		})
 		return ErrTooManyCardsLinked
 	}
 
-	err = pg.TxLinkWalletInfo(ctx, tx, ID, userDepositDestination, providerLinkingID, anonymousAddress, depositProvider)
+	// check the linking limit does not exceed what is appropriate
 	if err != nil {
-		return errorutils.Wrap(err, "unable to set an anonymous address")
+		sublogger.Error().Err(err).
+			Msg("failed to insert wallet_custodian due to db err checking linking limits")
+		return fmt.Errorf("failed to insert wallet custodian record due to db err checking linking limits: %w", err)
 	}
 
-	err = tx.Commit()
+	stmt = `
+		insert into wallet_custodian (
+			wallet_id, custodian, linking_id
+		) values (
+			$1, $2, $3
+		)
+		on conflict (wallet_id, custodian, linking_id) 
+		do update set disconnected_at=null, linked_at=now()
+		returning *
+	`
+
+	err = tx.Get(cl, stmt, cl.WalletID, cl.Custodian, cl.LinkingID)
 	if err != nil {
-		return err
+		sublogger.Error().Err(err).
+			Msg("failed to insert wallet_custodian")
+		return fmt.Errorf("failed to insert wallet custodian record: %w", err)
+	}
+	// update wallets with new deposit destination
+	stmt = `
+		update wallets set user_deposit_destination=$1 where id=$2
```

Comment:
if we're already updating wallets why not keep the other linking fields up to date?

Reply:
resolved: https://github.com/brave-intl/bat-go/pull/793/commits/dcbcd7ca13f72be4249e0760d01398e454c5bc66